### PR TITLE
Refactor payment status changes to use pre-save signal

### PR DESCRIPTION
**Pull request by oleast**
## What kind of a pull request is this?

- QA / Code Review

## Code Checklist

- [x] The code follows dotkom code style 
- [x] The code passes the defined tests
- [ ] I have added tests for the code I added (tests already cover this)
- [ ] I have provided documentation for the code I added
- [x] The code is ready to be merged

## (Possible) Breaking Changes

- [x] The changes are backwards compatible

## Description of changes

Use pre-save signals to handle changes in payment status.
Makes things a bit less bloated in models and handles side effects a bit more clearly?

### CODE REVIEW

File: apps/payment/signals.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        instance.user.change_saldo(instance.amount)
+
+        """ Pass the transaction to the next step, which is DONE """
+        instance.status = status.DONE
+
+    """ Handle when a transaction is being refunded by Stripe """
+    if instance.status == status.REFUNDED:
+        instance.user.change_saldo(-instance.amount)
+
+        """ Pass transaction to the next strip, which is REMOVED """
```
Comment (torjusti) [inline apps/payment/signals.py]: Seems stylistically a little weird to use docstrings instead of comments here?

File: apps/payment/signals.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        instance.status = status.DONE
+    elif instance.status == status.REFUNDED:
+        instance.refunded = True
+        instance.payment.handle_refund(instance)
+        instance.status = status.REMOVED
+
+
+@receiver(signal=pre_save, sender=PaymentTransaction)
+def handle_payment_transaction_status_change(sender, instance: PaymentTransaction, **kwargs):
+    # When a payment succeeds, ot should be stored to the DB """
```
Comment (torjusti) [inline apps/payment/signals.py]: You missed the closing quotations :stuck_out_tongue:

Comment (oleast) [reply apps/payment/signals.py]: :upside_down_face:

File: apps/payment/appconfig.py [hunk 3]
```diff
@@ -7,3 +7,6 @@ class PaymentConfig(AppConfig):
 
     def ready(self):
         super(PaymentConfig, self).ready()
+        # The following stops pycharm from nagging about unused import statement
+        # noinspection PyUnresolvedReferences
+        import apps.payment.signals  # noqa: F401
```
Comment (torjusti) [inline apps/payment/appconfig.py]: OK, so I have a more substantial comment this time.

From reading the docs, it seems like this `ready()` method can end up being called multiple times. When testing locally it seems to actually be the case as well, and it seems to cause the signal receivers to actually be registered twice as well. The receipt also seemingly gets sent twice.

Do you know anything about this behavior? Can we make this import only run once, somehow?

Comment (oleast) [reply apps/payment/appconfig.py]: I have no idea about it. I just did it as all the other apps did it

Comment (torjusti) [reply apps/payment/appconfig.py]: Sorry, I can't read. The import might run twice but the listeners only run once.

**Pull request merged**